### PR TITLE
Remove MojangAccounts

### PR DESCRIPTION
**Pull request by Doomsdayrs**
### Description of the Change

A heavy goodbye.

### Testing

- [x] Open Launcher
- [x] Add an account
- [x] Remove an account

Untested: Having a `MojangAccount`. But we [do not care](https://github.com/ATLauncher/ATLauncher/issues/906#issuecomment-2126058998) for that anymore.

### Related Issues

#906

### CODE REVIEW

File: src/main/java/com/atlauncher/managers/AccountManager.java
Comment (RyanTheAllmighty) [inline src/main/java/com/atlauncher/managers/AccountManager.java:1]: Can we change the `ACCOUNTS` behaviour subject be just MIcrosoftAccount now. I don't see Microsoft changing it and allowing different types of accounts after all the effort gone into moving to MIcrosoft accounts. Unless they sell Minecraft again, but I really don't think it's worth keeping all the casts around the place, let's just make it so the only place AbstractAccount class is used is in this file (just to keep backwards compatibility when reading accounts.json) and that's it

**Review by RyanTheAllmighty — CHANGES_REQUESTED** [review]
Can we also remove any references to AuthLib and remove it from the build dependencies as well?

**Comment by Doomsdayrs** [comment]
> Can we also remove any references to AuthLib and remove it from the build dependencies as well?

Please review this commit. Two dependencies the project used were from AuthLib, and I had to compensate. [33d4aaa](https://github.com/ATLauncher/ATLauncher/pull/907/commits/33d4aaab0f67a2f06365142fb98e639d091c3683)

File: src/main/java/com/atlauncher/managers/AccountManager.java
Comment (Doomsdayrs) [reply src/main/java/com/atlauncher/managers/AccountManager.java:1]: Completed via [91d7656](https://github.com/ATLauncher/ATLauncher/pull/907/commits/91d76563dcab35e6b40737e097229927ed262781).

Comment (RyanTheAllmighty) [reply src/main/java/com/atlauncher/managers/AccountManager.java:1]: Too quick 😮

File: src/main/java/com/atlauncher/data/AbstractAccount.java [hunk 1]
```diff
@@ -267,7 +266,7 @@ public String getUUIDNoDashes() {
      * @return The real UUID for this Account
      */
     public UUID getRealUUID() {
-        return (this.uuid == null ? UUID.randomUUID() : UUIDTypeAdapter.fromString(this.uuid));
+        return (this.uuid == null ? UUID.randomUUID() : UUID.fromString(this.uuid));
```
Comment (RyanTheAllmighty) [inline src/main/java/com/atlauncher/data/AbstractAccount.java:269]: So issue with this is that `this.uuid` is not a valid uuid. Mojang api's come back with a stripped UUID so for instance `672c66f60c56473cb220b95de0f52ae6`, so when it tried to run `UUID.fromString` it throws:

```
java.lang.IllegalArgumentException: Invalid UUID string: 672c66f60c56473cb220b95de0f52ae6
        at java.base/java.util.UUID.fromString1(Unknown Source)
        at java.base/java.util.UUID.fromString(Unknown Source)
        at com.atlauncher.data.AbstractAccount.getRealUUID(AbstractAccount.java:269)
```

Comment (Doomsdayrs) [reply src/main/java/com/atlauncher/data/AbstractAccount.java:269]: Oh god the horrors: https://stackoverflow.com/a/19399768

Comment (Doomsdayrs) [reply src/main/java/com/atlauncher/data/AbstractAccount.java]: [75a2106](https://github.com/ATLauncher/ATLauncher/pull/907/commits/75a21066f1c7f6950675249c86e902e62d9846bb) seems to do it.

**Review by RyanTheAllmighty — APPROVED** [review]
LGTM. Fingers crossed no issues comes up, but I'll do some good testing tonight

**Pull request merged**